app/controllers/components/course/discussion/topics_component.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+class Course::Discussion::TopicsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def self.display_name
+    I18n.t('components.discussion.topics.name')
+  end
+
+  def sidebar_items
+    main_sidebar_items + settings_sidebar_items
+  end
+
+  def settings
+    @settings ||= Course::Discussion::TopicSettings.new(current_course.settings(:discussion_topics))
+  end
+
+  private
+
+  def main_sidebar_items
+    [
+      {
+        key: :discussion_topics,
+        title: settings.title || t('course.discussion.topics.sidebar_title'),
+        weight: 5,
+        path: sidebar_path,
+        unread: unread_count
+      }
+    ]
+  end
+
+  def settings_sidebar_items
+    [
+      {
+        title: t('course.discussion.topics.sidebar_title'),
+        type: :settings,
+        weight: 4,
+        path: course_admin_topics_path(current_course)
+      }
+    ]
+  end
+
+  def sidebar_path
+    if can?(:manage, current_course)
+      pending_course_topics_path(current_course)
+    elsif current_course_user && current_course_user.teaching_assistant?
+      my_students_pending_course_topics_path(current_course)
+    else
+      course_topics_path(current_course)
+    end
+  end
+
+  def unread_count
+    if can?(:manage, current_course)
+      all_unread_count
+    elsif current_course_user && current_course_user.teaching_assistant?
```

Comment:
Use-case-wise, I agree with @weiqingtoh. 

As a TA for CS1010FC/X, I had both roles: (1) manage the course (2) tutor students. It would be more useful for me (as a course_manager) to see the pending number for my students rather than all students since I care about the assignments I have to grade and trust each tutor to grade theirs. I'm not sure how much this applies to other course_managers.